tests/test_quantization.py

```diff
@@ -37,3 +38,13 @@ def test_quantize(self, use_marlin: bool):
             model.save_pretrained(tmpdirname)
 
             model = AutoGPTQForCausalLM.from_quantized(tmpdirname, device="cuda:0", use_marlin=use_marlin)
+
+            # test compat: 1) with simple dict type 2) is_marlin_format
+            old_quantize_config = {
+                "bits": 4,
+                "group_size": 128,
+                "desc_act": False,
+                "is_marlin_format": use_marlin,
+            }
+            model = AutoGPTQForCausalLM.from_quantized(tmpdirname, device="cuda:0", quantize_config=old_quantize_config)
+            assert(isinstance(model.quantize_config, BaseQuantizeConfig))
```

Comment:
@fxmarty added test to show 1) weak dict quantize_config input with old `is_marlin_format` is supported 2) weak dict input is force converted into strong BaseQuantizeConfig